src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/FieldMapper.java

```diff
@@ -37,9 +38,8 @@
     AttributeValue apply(FieldMapping fieldMapping, AttributeValue unqualifiedAttribute) {
         return new AttributeValue(
             fieldPrefixFunction.apply(mtContext,
-                fieldMapping.getIndexType() == TABLE
-                    ? virtualTableName
-                    : fieldMapping.getVirtualIndexName(),
+                virtualTableName
+                + (fieldMapping.getIndexType() == SECONDARYINDEX ? SEPARATOR + fieldMapping.getVirtualIndexName() : ""),
```

Comment:
This is the actual fix.

Reply:
Why append the index name?

Reply:
Also, this is a breaking change, right?

Reply:
I don't know why I included in the index name.

Reply:
This is a breaking change for anyone who happened to use gsi's.

Reply:
fixed in 4910add08ec10862669e0089d59a222d4602de48